images/router/haproxy/conf/haproxy-config.template

```diff
@@ -151,6 +151,7 @@ backend be_tcp_{{$id}}
 
             {{ if eq $cfg.TLSTermination "reencrypt" }}
 backend be_secure_{{$id}}
+  mode http
```

Comment:
So when you reencrypt, HAProxy needs to think it's in http mode so that it processes the HTTP logic?  I see the SSL check on the backends - that's all it needs to know?